### PR TITLE
Add `LogTarget::AppDir` & `LogTarget::Webview`

### PR DESCRIPTION
**Pull request by JonasKruckenberg**
This adds the `AddDir` option to the `LogTarget` enum, it will acts basically the same as the `Folder` target but chooses the appropriate directory based on `tauri::api::path::app_dir`. 
This has the convenience that users don't need to interact with the unstable output of `tauri::generate_context` when configuring the plugin.

This also adds another target: `LogTarget::Webview` this emits all logs to the web webview windows through the `log://log` event. This event can be intercepted manually or directly piped into the web views console through the `attachConsole` function provided by the api package.

```js
import { attachConsole } from 'tauri-plugin-log-api'

const detach = await attachConsole()

// detach can be called to stop printing logs to the console
detach()
```

### CODE REVIEW

**Comment by JonasKruckenberg** [comment]
Actually I just realized I somehow merged some other stuff into this branch... Shall I just recreate this PR?

**Comment by lucasfernog** [comment]
I saw the webview target, you can just put everything on this PR since we didn't release this plugin yet.

File: webview-src/index.ts [hunk 1]
```diff
@@ -36,6 +36,33 @@ export async function error(message: string): Promise<void> {
   await log(LogLevel.Error, message)
 }
 
+interface RecordPayload {
+  level: number;
+  message: string;
+}
+
 export function attachConsole() {
```
Comment (JonasKruckenberg) [inline webview-src/index.ts:44]: I don't really like the name of this function🤔

Comment (lucasfernog) [reply webview-src/index.ts:44]: maybe `bridgeToConsole`? also maybe check the log level and redirect to console.log, console.warn, console.error etc

Comment (JonasKruckenberg) [reply webview-src/index.ts:44]: Is it not already checking that? I thought so 🤔

Comment (lucasfernog) [reply webview-src/index.ts:44]: oops :)

**Pull request merged**